platform/security/sts/security-sts-casclaimshandler/src/main/java/org/codice/ddf/security/claims/cas/CasClaimsHandler.java

```diff
@@ -0,0 +1,185 @@
+
+  private Map<String, String> attributeMap = new HashMap<>();
+
+  private List<String> supportedClaims = new ArrayList<>();
+
+  private List<String> supportedRealms;
+
+  private String realm;
+
+  @Override
+  public List<URI> getSupportedClaimTypes() {
+    List<URI> supportedClaimTypes = new ArrayList<>();
+
+    for (String claim : supportedClaims) {
+      try {
+        supportedClaimTypes.add(new URI(claim));
+      } catch (URISyntaxException e) {
+        LOGGER.error("Invalid supported claim: \"{}\"", claim);
+      }
+    }
+
+    return supportedClaimTypes;
+  }
+
+  @Override
+  public ProcessedClaimCollection retrieveClaimValues(
+      ClaimCollection claims, ClaimsParameters parameters) {
+    ProcessedClaimCollection claimsColl = new ProcessedClaimCollection();
+    Principal principal = parameters.getPrincipal();
+    Map<String, Object> additionalProperties = parameters.getAdditionalProperties();
+
+    if (additionalProperties != null) {
+      if (LOGGER.isTraceEnabled()) {
+        StringBuffer output = new StringBuffer();
+        String prefix = "\n";
+        output.append("CAS attributes returned: {");
+        for (Map.Entry<String, Object> entry : additionalProperties.entrySet()) {
+          output.append(prefix);
+          prefix = ", \n";
+          output.append(entry.getKey());
+          output.append("=");
+          output.append(entry.getValue());
+        }
+        output.append("\n}");
+        LOGGER.trace(output.toString());
+      }
+
+      claims
+          .stream()
+          .distinct()
+          .forEach(
+              claim -> {
+                URI claimType = claim.getClaimType();
+                String claimTypeString = claimType.toString();
+                String mappedClaimType =
+                    attributeMap.getOrDefault(claimTypeString, claimTypeString);
+                buildClaim(
+                    claimsColl, principal, claimType, additionalProperties.get(mappedClaimType));
+              });
+    }
+
+    if (LOGGER.isTraceEnabled()) {
+      StringBuffer output = new StringBuffer();
+      output.append("Claims map returned: {");
+      String prefix = "\n";
+      for (ProcessedClaim claim : claimsColl) {
+        output.append(prefix);
+        prefix = ", \n";
+        output.append(claim.toString());
+      }
+      output.append("\n}");
+      LOGGER.trace(output.toString());
+    }
+
+    return claimsColl;
+  }
+
+  private void buildClaim(
+      ProcessedClaimCollection claimsColl, Principal principal, URI claimType, Object value) {
+    if (value == null) {
+      return;
+    }
+
+    ProcessedClaim c = new ProcessedClaim();
+    c.setClaimType(claimType);
+    c.setPrincipal(principal);
+
+    // Multivalued attributes have values of the form "[val1, val2, ...]". Match such strings
+    // and return the substring within the brackets as group 1
+    Pattern p = Pattern.compile("^\\[(.*)\\]$");
```

Comment:
Should compile the pattern as a class field instead so we don't need to compile it many times.